MQClient/queue.py

```diff
@@ -52,32 +60,45 @@ def prefetch(self, val: int) -> None:
                 del self.raw_sub_queue
 
     @property
-    def raw_pub_queue(self) -> typing.Any:
+    def raw_pub_queue(self) -> Pub:
+        """Get publisher queue."""
         if not self._pub_queue:
             self._pub_queue = self._backend.create_pub_queue(self._address, self._name)
+
+        if not self._pub_queue:
+            raise Exception("Pub queue failed to be created.")
         return self._pub_queue
 
     @raw_pub_queue.deleter
     def raw_pub_queue(self) -> None:
+        logging.debug("Deleter Queue.raw_pub_queue")
         self._close_pub_queue()
 
-    def _close_pub_queue(self):
+    def _close_pub_queue(self) -> None:
         if self._pub_queue:
+            logging.debug("Closing Queue._pub_queue")
             self._pub_queue.close()
             self._pub_queue = None
 
     @property
-    def raw_sub_queue(self) -> typing.Any:
+    def raw_sub_queue(self) -> Sub:
+        """Get subscriber queue."""
         if not self._sub_queue:
-            self._sub_queue = self._backend.create_sub_queue(self._address, self._name, self._prefetch)
+            self._sub_queue = self._backend.create_sub_queue(
+                self._address, self._name, self._prefetch)
+
+        if not self._sub_queue:
+            raise Exception("Sub queue failed to be created.")
         return self._sub_queue
```

Comment:
Something feels a little weird about a getter creating the subscriber queue.
I don't have an alternate design to offer off the top of my head, though.

Reply:
(see above)